src/proc/syncer-processing-block.cpp

```diff
@@ -10,63 +10,98 @@
 
 namespace librealsense
 {
-    syncer_process_unit::syncer_process_unit( std::initializer_list< bool_option::ptr > enable_opts )
-        : processing_block("syncer"), _matcher((new timestamp_composite_matcher({})))
-        , _enable_opts( enable_opts.begin(), enable_opts.end() )
-    {
-        _matcher->set_callback([this](frame_holder f, syncronization_environment env)
+syncer_process_unit::syncer_process_unit( std::initializer_list< bool_option::ptr > enable_opts,
+                                          bool log )
+    : processing_block( "syncer" )
+    , _enable_opts( enable_opts.begin(), enable_opts.end() )
+{
+    auto f = [this, log]( frame_holder frame, synthetic_source_interface * source ) {
+        // if the syncer is disabled passthrough the frame
+        bool enabled = false;
+        size_t n_opts = 0;
+        for( auto & wopt : _enable_opts )
         {
-            std::stringstream ss;
-            ss << "SYNCED: ";
-            auto composite = dynamic_cast<composite_frame*>(f.frame);
-            for (int i = 0; i < composite->get_embedded_frames_count(); i++)
+            auto opt = wopt.lock();
+            if( opt )
             {
-                auto matched = composite->get_frame(i);
-                ss << matched->get_stream()->get_stream_type() << " " << matched->get_frame_number() << ", "<<std::fixed<< matched->get_frame_timestamp()<<" ";
+                ++n_opts;
+                if( opt->is_true() )
+                {
+                    enabled = true;
+                    break;
+                }
             }
+        }
+        if( n_opts && ! enabled )
+        {
+            get_source().frame_ready( std::move( frame ) );
+            return;
+        }
 
-            LOG_DEBUG(ss.str());
-            env.matches.enqueue(std::move(f));
-        });
-
-        auto f = [&](frame_holder frame, synthetic_source_interface* source)
         {
-            // if the syncer is disabled passthrough the frame
-            bool enabled = false;
-            size_t n_opts = 0;
-            for( auto& wopt : _enable_opts )
+            std::lock_guard< std::mutex > lock( _mutex );
```

Comment:
Why is this needed? We're in a ctor... who's gonna be using us before we return?